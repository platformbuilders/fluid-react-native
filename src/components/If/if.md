
# If 

## If a condition is true it will display the component

```js
import React from 'react';
import { Text } from 'react-native';
import { If } from '@platformbuilders/react-native-ui';

const Home: React.FC = () => {
   <If condition={true}>
      <Text>Hello</Text>
   </If>
  );
}

export default Home;
```


| Name  | Required | Type | Default Value | Description 
| ------------- | ------------- | ------------- |------------- |------------- |
| condition | true | boolean |  |  |
