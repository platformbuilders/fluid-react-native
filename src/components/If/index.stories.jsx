import { storiesOf } from '@storybook/react-native';
import ifMarkdown from './if.md';

storiesOf('If', module)
  .addParameters({
    readme: {
      content: ifMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => {});
