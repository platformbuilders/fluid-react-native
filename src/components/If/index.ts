interface Props {
  condition: boolean;
  children: any;
}

const If = ({ condition, children }: Props): any =>
  condition ? children : null;

export default If;
