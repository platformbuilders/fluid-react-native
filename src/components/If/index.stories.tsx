import React from 'react';
import { Text } from 'react-native';
import type { Meta } from '@storybook/react';
import If from '.';

const meta: Meta<typeof If> = {
  title: 'If',
  component: If,
  argTypes: {
    condition: { control: { type: 'boolean' } },
  },
  args: {
    condition: true,
    children: <Text>Texto de exemplo</Text>,
  },
  parameters: {
    notes: 'Componente para renderização condicional de elementos.',
  },
};

export default meta;

export const Default = {};
