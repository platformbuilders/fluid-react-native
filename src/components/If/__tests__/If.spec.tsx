import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import If from '..';

describe('<If />', () => {
  it('should match snapshot when truthy', () => {
    const render = renderer.create(
      <If condition>
        <Text>Test</Text>
      </If>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot when false', () => {
    const render = renderer.create(
      <If condition={false}>
        <Text>Test</Text>
      </If>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render the component when the condition is truthy', () => {
    const render = renderer.create(
      <If condition>
        <Text>Test</Text>
      </If>,
    );

    expect(JSON.stringify(render.toJSON())).toBe(
      JSON.stringify({
        type: 'Text',
        props: {},
        children: ['Test'],
      }),
    );
  });

  it('should not render the component when the condition is falsy', () => {
    const render = renderer.create(
      <If condition={false}>
        <Text>Test</Text>
      </If>,
    );

    expect(render.toJSON()).toBe(null);
  });
});
