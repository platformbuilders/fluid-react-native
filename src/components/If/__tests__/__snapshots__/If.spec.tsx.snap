// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<If /> should match snapshot when false 1`] = `null`;

exports[`<If /> should match snapshot when truthy 1`] = `
<Text>
  Test
</Text>
`;
